app/javascript/components/Body.js

```diff
@@ -0,0 +1,72 @@
+import React from "react"
+import ProductList from "./ProductList";
+import AddProduct from "./AddProduct";
+import * as moment from "moment"
+
+class Body extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            products: []
+        };
+        this.handleSubmit = this.handleSubmit.bind(this)
+        this.addNewProduct = this.addNewProduct.bind(this)
+        this.handleDelete = this.handleDelete.bind(this)
+    }
+    handleSubmit(name, expiry_date, quantity) {
+        var expiry_date_formatted = moment(expiry_date).format('YYYY-MM-DD')
+        var body = JSON.stringify({product: {name: name, expiry_date: expiry_date_formatted, quantity: quantity}})
+        var token = document.querySelector('meta[name="csrf-token"]').getAttribute('content')
+        fetch('/products', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+                'X-CSRF-TOKEN': token
+            },
+            body: body
+        }).then((response) => {
```

Comment:
Have you looked into using `async/await` for these requests? I find them more readable and I think they're also the "latest" way of doing async calls.